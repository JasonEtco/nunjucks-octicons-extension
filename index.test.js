const nunjucks = require('nunjucks')
const Extension = require('.')

describe('nunjucks-octicons-extension', () => {
  let env

  beforeEach(() => {
    env = nunjucks.configure()
    env.addExtension('octicon', Extension)
  })

  it('renders the correct octicon', () => {
    const actual = env.renderString('{% octicon "hubot" %}')
    expect(actual).toMatchSnapshot()
  })

  it('renders the correct octicon with options', () => {
    const actual = env.renderString('{% octicon "hubot", height="64", width="64", class="m-1" %}')
    expect(actual).toMatchSnapshot()
  })

  it('throws if the request octicon does not exist', () => {
    try {
      env.renderString('{% octicon "pizza" %}')
    } catch (e) {
      expect(e).toMatchSnapshot()
    }
  })
})
