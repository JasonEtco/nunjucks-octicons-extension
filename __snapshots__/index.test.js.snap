// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nunjucks-octicons-extension renders the correct octicon 1`] = `"<svg version=\\"1.1\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" class=\\"octicon octicon-hubot\\" aria-hidden=\\"true\\"><path fill-rule=\\"evenodd\\" d=\\"M0 8a8 8 0 1116 0v5.25a.75.75 0 01-1.5 0V8a6.5 6.5 0 10-13 0v5.25a.75.75 0 01-1.5 0V8zm5.5 4.25a.75.75 0 01.75-.75h3.5a.75.75 0 010 1.5h-3.5a.75.75 0 01-.75-.75zM3 6.75C3 5.784 3.784 5 4.75 5h6.5c.966 0 1.75.784 1.75 1.75v1.5A1.75 1.75 0 0111.25 10h-6.5A1.75 1.75 0 013 8.25v-1.5zm1.47-.53a.75.75 0 011.06 0l.97.97.97-.97a.75.75 0 011.06 0l.97.97.97-.97a.75.75 0 111.06 1.06l-1.5 1.5a.75.75 0 01-1.06 0L8 7.81l-.97.97a.75.75 0 01-1.06 0l-1.5-1.5a.75.75 0 010-1.06z\\"></path></svg>"`;

exports[`nunjucks-octicons-extension renders the correct octicon with options 1`] = `"<svg version=\\"1.1\\" width=\\"64\\" height=\\"64\\" viewBox=\\"0 0 24 24\\" class=\\"octicon octicon-hubot m-1\\" aria-hidden=\\"true\\"><path d=\\"M0 13C0 6.373 5.373 1 12 1s12 5.373 12 12v8.657a.75.75 0 01-1.5 0V13c0-5.799-4.701-10.5-10.5-10.5S1.5 7.201 1.5 13v8.657a.75.75 0 01-1.5 0V13z\\"></path><path d=\\"M8 19.75a.75.75 0 01.75-.75h6.5a.75.75 0 010 1.5h-6.5a.75.75 0 01-.75-.75z\\"></path><path fill-rule=\\"evenodd\\" d=\\"M5.25 9.5a1.75 1.75 0 00-1.75 1.75v3.5c0 .966.784 1.75 1.75 1.75h13.5a1.75 1.75 0 001.75-1.75v-3.5a1.75 1.75 0 00-1.75-1.75H5.25zm.22 1.47a.75.75 0 011.06 0L9 13.44l2.47-2.47a.75.75 0 011.06 0L15 13.44l2.47-2.47a.75.75 0 111.06 1.06l-3 3a.75.75 0 01-1.06 0L12 12.56l-2.47 2.47a.75.75 0 01-1.06 0l-3-3a.75.75 0 010-1.06z\\"></path></svg>"`;

exports[`nunjucks-octicons-extension throws if the request octicon does not exist 1`] = `
[Template render error: (unknown path)
  Error: Octicon pizza does not exist]
`;
